### ISSUE TITLE
Non existing url on russian lang redirected to main page

### ISSUE THREAD

**Issue by akudryav** [phalcon/cphalcon#10544]
If you write some blabla url using russian like this: site.ext\проверка no 404 error happens.
Tested on Invo and Vokuro applications.

**Comment by andresgutierrez** [comment]
Do you have namespaces, controllers or actions with names in Russian?

**Comment by sergeyklay** [comment]
@andresgutierrez 
Russia uses IPv6. This allows you to use a URLs with national symbols encoding

**Comment by stibiumz** [comment]
Also occurs with Latin-1 chars
https://invo.phalconphp.com/eñe -> Found
https://invo.phalconphp.com/ene -> Not found (unauthorized)

@sergeyklay this is IDN (not the same as an url path), and it is also supported by IPv4 addresses

**Comment by andresgutierrez** [comment]
@stibiumz you have controllers in russian?

**Comment by stibiumz** [comment]
I have not. Why does that matter?

**Comment by roman-kulish** [comment]
@sergeyklay  IPv4 vs IPv6 does not matter. 

For Russian it is either "%D0%BF%D1%80%D0%BE%D0%B2%D0%B5%D1%80%D0%BA%D0%B0" is not decoded back to "проверка" or preg_match in Route::compilePattern() should use Unicode flag.

For "eñe" vs "ene" they are two different words and I think some kind of a http://php.net/manual/en/class.normalizer.php should be used to process is or a special route to cover both :)

**Comment by akudryav** [comment]
There is no any controllers or actions with russian names, seems that this is routing bug.

**Comment by stibiumz** [comment]
@roman-kulish I know, the example was for illustrating, what I mean is that 'ene' route is recognized as not found as when with 'eñe' simply goes to index page, it's strange.

About matching `eñe` as if were `ene`, can be done with Transliterator, Normalizer is for character de-/composition

**Comment by andresgutierrez** [comment]
I think you can create the router using the #U operator:

```php
$router = new \Phalcon\Mvc\Router(false);

$router->add("#^/([\w\\_\\-]+)[/]{0,1}$#U", [
    "controller" => 1
]);

$router->add("#^/([\w0-9\\_\\-]+)/([a-zA-Z0-9\\.\\_]+)(/.*)*$#U", [
      "controller" => 1,
      "action"  =>  2,
      "params"  =>  3
]);
```

**Comment by hakimio** [comment]
I can see this bug too. I think by default Phalcon should just give 404 page in case non-latin words are used and no route can be matched instead of just totally disregarding the non-latin path.

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
